tajo-client/src/main/java/org/apache/tajo/client/CatalogAdminClientImpl.java

```diff
@@ -200,12 +231,17 @@ public TableDesc getTableDesc(final String tableName) throws SQLException {
       throw new RuntimeException(e);
     }
 
-    throwIfError(res.getState());
+    if (isThisError(res.getState(), Errors.ResultCode.UNDEFINED_TABLE)) {
+      throw new UndefinedTableException(res.getState());
+    }
+
+    ensureOk(res.getState());
     return CatalogUtil.newTableDesc(res.getTable());
   }
 
   @Override
-  public List<FunctionDescProto> getFunctions(final String functionName) throws SQLException {
+  public List<FunctionDescProto> getFunctions(final String functionName)
+      throws AmbiguousFunctionException, UndefinedFunctionException {
```

Comment:
As I commented above, it would be better if ```getFunctions``` returns an empty list rather than throwing an ```UndefinedFunctionException``` when there is no function of the given name.
In addition, this method should return a list of the functions of the given name. So, ```AmbiguousFunctionException``` is inappropriate.